jbpm-wb-case-mgmt/jbpm-wb-case-mgmt-client/src/main/java/org/jbpm/workbench/cm/client/roles/CaseRoleItemView.java

```diff
@@ -16,64 +16,218 @@
+    Div usersDiv;
 
     @Inject
-    @DataField("actions-dropdown")
-    Div actions;
+    @DataField("line-groups")
+    ItemsLine groupsItemsLine;
 
     @Inject
-    @DataField("actions-items")
-    UnorderedList actionsItems;
+    @DataField("div-groups")
+    Div groupsDiv;
+
+    @Inject
+    ConfirmPopup confirmPopup;
+
+    @Inject
+    protected TranslationService translationService;
+
+    @Inject
+    @AutoBound
+    private DataBinder<CaseRoleAssignmentSummary> caseRoleAssignmentSummary;
+
+    boolean listenLoadLineEvent=true;
 
     @Override
     public HTMLElement getElement() {
         return listGroupItem;
     }
 
-    public void setRoleName(final String roleName) {
-        this.roleName.setInnerHTML(roleName);
+    @Override
+    public CaseRoleAssignmentSummary getValue() {
+        return caseRoleAssignmentSummary.getModel();
     }
 
-    public void setName(final String userName) {
-        this.name.setInnerHTML(userName);
+    @Override
+    public void setValue(final CaseRoleAssignmentSummary model) {
+        this.caseRoleAssignmentSummary.setModel(model);
+        removeCSSClass(actions, "dropup");
+        if (CaseRolesPresenter.CASE_OWNER_ROLE.equals(model.getName())) {
+            setOwnerAssignment();
+        } else {
+            boolean hasAssignment = model.getUsers().size() + model.getGroups().size() > 0;
+            addAction(new CaseRolesPresenter.CaseRoleAction() {
+                @Override
+                public String label() {
+                    if (hasAssignment) {
+                        return translationService.format(EDIT);
+                    }
+                    return translationService.format(ASSIGN);
+                }
+
+                @Override
+                public boolean isEnabled() {
+                    return true;
+                }
+
+                @Override
+                public void execute() {
+                    presenter.editAction(model, model.getUsers(), model.getGroups());
+                }
+            });
+            addAction(new CaseRolesPresenter.CaseRoleAction() {
+                @Override
+                public String label() {
+                    return translationService.format(REMOVE_ALL_ASSIGNMENTS);
+                }
+
+                @Override
+                public boolean isEnabled() {
+                    return hasAssignment;
+                }
+
+                @Override
+                public void execute() {
+                    confirmPopup.show(translationService.format(REMOVE_ASSIGNMENT),
+                            translationService.format(REMOVE),
+                            translationService.format(REMOVE_ALL_USERS_GROUPS_FROM_ROLE, model.getName()),
+                            () -> presenter.storeRoleAssignments(model.getName(), Collections.emptyList(),
+                                    Collections.emptyList(), model.getUsers(), model.getGroups()));
+                }
+            });
+        }
+    }
```

Comment:
Scenario:
1. Role doesn't have any assignments.
2. Click Edit to add role assignment.
3. Add some user/group then cancel the action (either by clicking Cancel or closing the dialog) 
4. Open actions dropdown for that role again (screenshot below)
![screenshot from 2017-03-28 18-19-31](https://cloud.githubusercontent.com/assets/19285782/24415863/28551a56-13e3-11e7-9173-b1acbd741e1c.png)